### PR TITLE
Use 64-bit nonce

### PR DESCRIPTION
**Pull request by akorosmezey**
Needed for 
[PT-152672154](https://www.pivotaltracker.com/story/show/152672154) - 64-bits PoW nonce
[PT-152915267](https://www.pivotaltracker.com/story/show/152915267) - PoW nonce validation

### CODE REVIEW

File: apps/aecore/src/aec_pow.erl [hunk 1]
```diff
@@ -131,11 +131,11 @@ target_to_difficulty(Th) ->
 
 -spec pick_nonce() -> integer().
 pick_nonce() ->
-    rand:uniform(?NONCE_RANGE) band 16#7fffffff.
+    rand:uniform(?NONCE_RANGE) band ?MAX_NONCE.
 
 -spec next_nonce(integer()) -> integer().
 next_nonce(N) ->
```
Comment (hanssv) [inline apps/aecore/src/aec_pow.erl]: If we know that N =< ?MAX_NONCE it is more efficient to write this function as
```erlang
next_nonce(?MAX) -> 0;
next_nonce(N) -> N + 1. 
```
but that might not always be the case...

File: apps/aecore/src/aec_pow_cuckoo.erl [hunk 2]
```diff
@@ -20,8 +20,13 @@
 -compile([export_all, nowarn_export_all]).
 -include_lib("eunit/include/eunit.hrl").
 -endif.
+-include("pow.hrl").
 
--define(DEFAULT_CUCKOO_ENV, {"mean30", "-t 5", 30}). %% -m 7 caused crash
+-define(DEFAULT_CUCKOO_ENV, {"mean30", "-t 1", 30}). %% -m 7 caused crash
```
Comment (hanssv) [inline apps/aecore/src/aec_pow_cuckoo.erl]: Is this change intentional? There are issues with running cuckoo miner multi-threaded but I thought we concluded that for the actual miner (running in a separate thread) it will be better to use (5) threads... In some of our tests we'd like to run it single-threaded though!

File: apps/aecore/test/aec_mining_tests.erl [hunk 3]
```diff
@@ -174,4 +175,16 @@ cleanup(_, PoWMod) ->
             ok
     end.
 
+mine(_, _, 0) ->
```
Comment (hanssv) [inline apps/aecore/test/aec_mining_tests.erl]: Unused code?

Comment (lucafavatella) [reply apps/aecore/test/aec_mining_tests.erl]: Yes.
Also line 118. I understand this was draft code for actually finding nonce that has a solution for updating test.

File: apps/aecore/src/aec_pow_sha256.erl [hunk 4]
```diff
@@ -46,7 +47,9 @@ generate(Data, Target, Nonce) ->
 -spec verify(Data :: aec_sha256:hashable(), Nonce :: integer(),
              Evd :: aec_pow:pow_evidence(), Target :: aec_pow:sci_int()) ->
                     boolean().
-verify(Data, Nonce, Evd, Target) when Evd == no_value ->
+verify(Data, Nonce, Evd, Target) when Evd == no_value,
+                                      Nonce >= 0,
```
Comment (lucafavatella) [inline apps/aecore/src/aec_pow_sha256.erl]: I see this hardcodes expectation that nonce is in valid range. Thanks.

Comment (lucafavatella) [reply apps/aecore/src/aec_pow_sha256.erl]: This is part of PT-152915267.

**Comment by lucafavatella** [comment]
I see that one CI build passed and one [CI build failed](https://travis-ci.org/aeternity/epoch/jobs/302686519) but I understand it is not related to this PR. I have seen this in other rare occasions.

File: apps/aecore/src/aec_pow_cuckoo.erl (hunk 2 above)
Comment (lucafavatella) [reply apps/aecore/src/aec_pow_cuckoo.erl]: @akorosmezey If this is for making tests stable, I believe it is not needed thanks to https://github.com/aeternity/epoch/commit/aba0fda1f7dec68bdd74a8776c9286dbf1662e44 (included in master hence in this PR). Please elaborate.

File: config/swagger.yaml [hunk 5]
```diff
@@ -240,7 +240,7 @@ definitions:
         format: int64
       nonce:
         type: integer
-        format: int64
+        format: uint64
```
Comment (lucafavatella) [inline config/swagger.yaml]: I see this definition brings the HTTP API spec (swagger.yaml) in line with the documentation of the nonce in the header as `u64` in https://github.com/aeternity/epoch/wiki/Block-Header - thanks.

**Comment by lucafavatella** [comment]
I see the [serialization for hash](https://github.com/aeternity/epoch/blob/0e508233e143d94fd361aa116b4584153c69451b/apps/aecore/src/aec_headers.erl#L171) (`<< ... (H#header.nonce):64 ... >>`) is 64 bits [unsigned integer](http://erlang.org/doc/reference_manual/expressions.html#id83099) - ok.

File: apps/aecore/src/aec_pow_cuckoo.erl (hunk 2 above)
Comment (akorosmezey) [reply apps/aecore/src/aec_pow_cuckoo.erl]: Thanks for spotting, Fixed in next commit 6a49d96

**Comment by akorosmezey** [comment]
A new PR: #380 is created which is rebased on current master
